src/file_watching.jl

```diff
@@ -1,56 +1,153 @@
+"A watched file; construct with its path as single argument"
+mutable struct WatchedFile
+    path::String
+    mtime::Float64
+    WatchedFile(path::String) = new(path, mtime(path))
+end
+
+"Check if a `WatchedFile` has changed"
+has_changed(wf::WatchedFile) = mtime(wf.path) > wf.mtime
+
+"Set the current state of a `WatchedFile` as unchanged"
+set_unchanged!(wf::WatchedFile) = wf.mtime = mtime(wf.path)
+
 """
-    FileWatcher
+    SimpleWatcher([callback];sleeptime::Float64=0.1)
+
+A simple file watcher. You can specify a callback function, receiving the
+path of each file that has changed as a string, at construction or later
+by the API function described below. The `sleeptime` is the time waited
+between to runs of the loop looking for changed files.
 
-Contains a list of watched files and a callback function that should be applied in the case of file
-events.
+# API functions
+TBD, where `w` is a `SimpleWatcher`. Main API functions (commonly used):
+
+- `start(w)`
+- `stop(w)`
+- `set_callback(w, fcn::Function)`
+- `watch_file(w, path::String)`
+
+Further API functions (probably never used in a normal use case):
+
+- `isrunning(w)`
+- `is_file_watched(w, path::String)`
 """
-struct FileWatcher
-    watched_files::Dict{String,Task}
-    callback::Function
-    FileWatcher(cb::Function) = new(Dict{String,Task}(), cb)
+mutable struct SimpleWatcher
+    callback::Union{Nothing,Function}
+    task::Union{Nothing,Task}
+    sleeptime::Float64
+    filelist::Vector{WatchedFile}
+    SimpleWatcher(fcn::Union{Nothing,Function}=nothing; sleeptime::Float64=0.2) = new(fcn,nothing,max(0.1,sleeptime),Vector{WatchedFile}())
```

Comment:
maybe write `callback` everywhere instead of `fcn` as it would otherwise be confusing. Also you probably want to have the default sleep time to be less than 0.2.

Reply:
Ok, will set default sleeptime to 0.1 and minimum to 0.05. And change `fcn` to `callback`. It's the details that count as well ;-)